pydantic/validate_call_decorator.py

```diff
@@ -59,9 +60,17 @@ def validate(function: AnyCallableT) -> AnyCallableT:
         def wrapper_function(*args, **kwargs):
             return validate_call_wrapper(*args, **kwargs)
 
+        @functools.wraps(function)
+        async def async_wrapper_function(*args, **kwargs):
+            return await validate_call_wrapper(*args, **kwargs)
+
         wrapper_function.raw_function = function  # type: ignore
+        async_wrapper_function.raw_function = function  # type: ignore
 
-        return wrapper_function  # type: ignore
+        if inspect.iscoroutinefunction(function):
+            return async_wrapper_function  # type: ignore
+        else:
+            return wrapper_function  # type: ignore
```

Comment:
Let's add the function definitions (`wrapper_function` and `async_wrapper_function`) to the relevant parts of this conditional. Can also move the `.raw_function` assignments there as well.

Reply:
ok